src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1098,15 +1089,15 @@ impl LinkCollector<'_, '_> {
             return None;
         }
 
-        let key = ResolutionInfo {
-            module_id,
-            dis: disambiguator,
-            path_str: path_str.to_owned(),
-            extra_fragment,
-        };
-        let diag =
-            DiagnosticInfo { item, dox, ori_link: &ori_link, link_range: link_range.clone() };
-        let (mut res, mut fragment) = self.resolve_with_disambiguator_cached(key, diag)?;
+        let (mut res, mut fragment) = self.resolve_with_disambiguator_cached(
+            ResolutionInfo {
+                module_id,
+                dis: disambiguator,
+                path_str: path_str.to_owned(),
+                extra_fragment,
+            },
+            DiagnosticInfo { item, dox, path_str: &ori_link, link_range: link_range.clone() },
```

Comment:
nit:
```suggestion
            DiagnosticInfo { path_str: &ori_link, ..diag_info },
```

Reply:
I don't think it'll work since DI is not Copy